### PR TITLE
DOC: contributor incubator

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

Adds prose to docs about the "incubator" (nee `#tidepool`) discussed on the call.

The purpose of this channel is that, operating on the theory than the hardest PR to a project is the first, is to provide some mentoring / support to people who want to become regular contributors but do not know where to start.

The other issue I think we have in the user -> first PR -> multiple PR -> reviewer pipeline as that we can not tell who of our first time contributors wants to become a regular contributor and who is doing a drive-by PR to scratch their itch.   This is a path where we are asking people to self-identify as wanting to be come regular contributors.

### CODE REVIEW

**Review by jklymak — APPROVED** [review]
Seems good, so long as that channel exists ;-)

**Comment by tacaswell** [comment]
I was holding off creating the channel until after we had merged this.

**Comment by timhoffm** [comment]
No offense meant, but I find the first two sentences rather intimidating. Listing reasons why one could be intimidated is counter-productive. I'd just do something like:

> If you are interested in becoming a regular contributor to Matplotlib, but don't know where to start or feel insecure about it, you can join our non-public communication channel for new contributors. To do so, please go to gitter and ask to be added to '#incubator'. *[original text from here on]* This is a private gitter room moderated by core Matplotlib developers where you can get guidance and support for your first few PRs. This is a place you can ask questions about anything: how to use git, github, how our PR review process works, technical questions about the code, or get "pre-review" on your PR.

**Comment by QuLogic** [comment]
I agree with this; giving a number to it (10M / month) is even more intimidating.

**Comment by jklymak** [comment]
I also agree.  Just wanted to get the thing rolling, but I don't think a PR with @timhoffm wording would be at all remiss.

**Pull request merged**
